### DISCUSSION TITLE
Can this stop being suggested?

### DISCUSSION THREAD

**Discussion by ashyashton** [Monika-After-Story/MonikaModDev#9723] [general - submod / external content]
I feel like, a lot of the time, someone asks if something can be added and the devs just say "get a submod lol"

One: In many cases, the submod doesn't exist. Not everyone is capable of coding, but that doesn't invalidate the fact that they should feel represented, so the rebuttal of "make it yourself" doesn't really work either.

Two: More importantly, _**you guys don't officially support submods.**_ Despite this, though, you routinely suggest people download them.

I have seen many people post here asking for help, only to be told that it's an issue with a submod so they need to go there for help. Let me be clear: I do not disagree with this policy. But you cannot suggest a player do something, then refuse to help when the action _you_ suggested backfires. That is not fair for devs to do. That would be like if someone was upset they missed an event, and someone on the team suggests changing the computer time to get to experience it, only to refuse to provide support because they time traveled and messed something up.

I get that not everything can be added to the mod. I have no problem with this. I'm not saying add the ideas anyways even though you don't want to/don't have the resources/etc. But please stop using nonexistent submods that you wouldn't provide support for as a solution to problems. It is not a fair solution; it's a cop-out to avoid explaining why you won't add something.

**Comment by Wingdinggaster656** [comment]
> More importantly, you guys don't officially support submods.

Actually not. Actually, MAS do support submods.
MAS has a relatively complete Submod access system and an interface for users to uniformly manage Submod.
Of course, this system isn't perfect, since it cannot perfectly let the writer of Submod override some events, which is a fact - but MAS still supports Submod after all.

**Comment by ashyashton** [comment]
@Wingdinggaster656 You are misunderstanding what I mean "support." I don't mean that you can't install them; I mean what I said in the post you responded to.

> I have seen many people post here asking for help, only to be told that it's an issue with a submod so they need to go there for help.

I mean support as in when people talk about tech support. I _explicitly_ said this, as quoted here, so I'm not sure why you're twisting it to mean a different form of the word support when I explicitly stated I meant the other form. Just to make the mod look better? Like, as opposed to addressing the issue, you just twist what's being said about the issue to brush it off?

While you can add submods, support is not provided if there is an issue with them. Saying "well technically you can add them" doesn't change that. I'm not sure what you thought you were achieving with that argument.

**Comment by Booplicate** [comment]
I'm not sure what's the point of this. Issues are for bug reports and mod suggestions. This is neither.

> More importantly, you guys don't officially support submods.

You should've clarified there what exactly you mean.

We indeed don't provide tech support for the code we didn't write. Submods are a opt-in feature. It's not about being able to *install*, it's about being able to *make* a submod. We have a public API for that and we fully support it.
An analogy would be translations, we support *making* them, but it's the creator responsibility to make them compatible and safe to use, provide tech support.

> That would be like if someone was upset they missed an event, and someone on the team suggests changing the computer time to get to experience it, only to refuse to provide support because they time traveled and messed something up.

Changing time like that is expected to break the mod, but using a submod shouldn't cause any data issues.

> in many cases, the submod doesn't exist. Not everyone is capable of coding

Many people learn how to code by making submods, actually. It's not for everyone, in that case those people can commission other people to make submods for them.

> But please stop using nonexistent submods that you wouldn't provide support for as a solution to problems. It is not a fair solution; it's a cop-out to avoid explaining why you won't add something.

We don't suggest to use specific submods - we don't write nor test them. But if something cannot be added to the base game, people can do research and find/make submods to fill the niche. It's the only solution in that case.

**Comment by ashyashton** [comment]
> I'm not sure what's the point of this. Issues are for bug reports and mod suggestions. This is neither.

It's a suggestion for the GitHub repository itself. I'm not aware of a space for that _other_ than the GitHub repository itself. I appreciate it being changed to a discussion; I would have set it up as such myself if I knew how to.

> You should've clarified there what exactly you mean.

I provided a direct example literally on the next line. I don't know how else I could have explained it; I stated the problem and gave an example. Please feel free to tell me how I should have said it.

> using a submod shouldn't cause any data issues.

Okay, but they _can_ cause issues, be it data issues or just plain crashing. **If it can cause any issues, and you will not help with said issues, you should not be suggesting it.**

> in that case those people can commission other people to make submods for them.

So on the incredibly short list of people who deserve to be represented, we have:
1. People who agree with devs and have the same feelings as them
2. Anyone who can code
3. People with money

Am I getting it right? Or am I missing something here? Feel free to correct me if I got that list wrong, but at the moment it feels pretty accurate.

**Reply by Booplicate** [reply]
> If it can cause any issues, and you will not help with said issues, you should not be suggesting it.

Generally any program can cause issues, following that logic nobody should suggest using anything at all.

We are not suggesting to use the submods that have known issues. Furthermore, we are not even suggesting to use an exact submod at all. All we do is saying that a feature isn't yet (or cannot be) in MAS, if you need it, look into submods. The API was created exactly for that purpose.
In the end *you* are finding and installing the submod of your choice on your computer, not we. You can check the code, reviews, changelogs, readme, make backups before using it.
After all it's an opt-in feature: don't want to deal with all that, don't use it. Or if you're making it yourself, then test your code before using it.

> So on the incredibly short list of people who deserve to be represented, we have

Your words, not mine. I'm not even sure what that all has to do with submods.

Back to this:

> they should feel represented

Sure, but that's too much of an expectation from a free mod for a free game made purely on inspiration. We physically cannot pleasure everyone and fulfill everyone's dreams. Sorry.

> I mean, the whole reason it isn't being added is because the dev team themselves finds it too difficult.

It's not just "too difficult." If that was the only our project, we would be able to focus on it. But we have other things in priority and we can't add another complex project ahead of everything because somebody wants so. Managing projects, deadlines, and current development direction is our responsibility.
Also reminder, while all suggestions and additions are carefully reviewed and considered, the dev team still has the final decision and sets the reason behind it.

> But can the submod API be used to add holidays?

Technically, yes. It can be used for any kind of submods. Another thing is, it's hard, so I, myself, am not expecting most users to do that.

> If yes, then why not use it for that issue?

You can use the submod framework for that issue if you have the time and skills to do that work.
Or you can commission it.
Or you can wait until we get our hands on that.

**Reply by ashyashton** [reply]
> Generally any program can cause issues, following that logic nobody should suggest using anything at all.

Wow I love when the devs completely ignore half of the sentence they're responding to! I said "and you will not help with said issues." Can devs of this mod _please_ stop cherrypicking arguments? It's immature and it makes it very hard to have respect for the people who are supposed to be in charge.

>Another thing is, it's hard, so I, myself, am not expecting most users to do that.

Then why does it keep being suggested? No one is addressing the core issue, but this right here is it: You _don't_ expect people do make these submods. You want to brush people off with a solution you _know_ doesn't work in most cases, but hey technically you suggested something so it's fine!

> You can use the submod framework for that issue if you have the time and skills to do that work.
Or you can commission it.
Or you can wait until we get our hands on that.

So you agree that only people with skills, money, or the people who agree with the devs have the right to be represented then?

**Comment by ashyashton** [comment]
I'd also like to elaborate on the first issue, the one regarding not everyone being able to make the submod if it doesn't exist, by giving an example. This example is very recent; it is in #9716

The first reply, from dreamscached (not tagging to avoid annoying people) is as follows:
> Holidays not celebrated worldwide are (imo) less likely to be added for everyone, but this is changeable with submods.

Later in the replies, other devs say the folowing:

Booplicate says:
> New holidays are unlikely to happen because of their complexity and overlapping with other events (player birthday hell), it'd take tremendous amount of time to add a big event like that.

Later, multimonika says:
> In addition to this, we'd also need to rework the way we handle autoload checks and the like to deal with events in a more modular fashion. [...] but ultimately a framework like this too, would take time and as well significant development.

So... How is "add it in a submod" a valid suggestion? I mean, the whole reason it isn't being added is because the dev team themselves finds it too difficult. Yet it's treated as if the solution is for one single player to add it in on their own, as if it were the easier option.

Now, you claim that you would provide full support for those using the submod API to make a submod. But can the submod API be used to add holidays? If yes, then why not use it for that issue? If no, then how is it fair to suggest it in that scenario?

**Reply by dreamscached** [reply]
You can make anything with the mod API, MAS events, holidays, and everything uses the same API submod can use.

**Reply by dreamscached** [reply]
> So... How is "add it in a submod" a valid suggestion? I mean, the whole reason it isn't being added is because the dev team themselves finds it too difficult. Yet it's treated as if the solution is for one single player to add it in on their own, as if it were the easier option.

They said about reworking an existing framework *for better convenience* not something player *need* to implement themselves.

**Reply by ashyashton** [reply]
That's a cop-out and you know it. They were very heavily implying that they _need_ to do it in order to add more holidays. Yet somehow it's fine that end-users have to do it the other way? You and I both know that you're just using an excuse at this point.

**Reply by ThePotatoGuy** [reply]
>They were very heavily implying that they need to do it in order to add more holidays.

we don't *need* to add a framework to add more holidays, but we also don't need to write good code that is easy for people later to extend. Holidays as currently implemented are like a stack of books of various shapes and sizes. Adding another holiday is like adding another oddly-shaped book to that stack, and the taller it gets, the more likely it falls (aka the game breaks somewhere). Making a framework is like designing a stackable case that can hold each book regardless of its shape and size so its less likely to fall.

>Yet somehow it's fine that end-users have to do it the other way

End-users would just have to find the desired submod. A better framework for holidays is geared toward submod creators so they won't have to suffer the same book-stacking headaches that we would suffer adding another event.

**Comment by dreamscached** [comment]
I personally don't see just *any* valid reason for anyone to stop suggesting using submods. Analogies are not valid arguments in a discussion but I can't imagine someone of the fanbase of (for instance) Minecraft being annoyed that game lacks X and then someone suggests using Y mod that adds wanted functionality.

Do game developers support *modding*? Yes, Minecraft has Forge on which (iirc) some game developers have been involved in development of.
Do game developers support *mods*? Nope. It's not their responsibility to manage and maintain code they didn't write but did their best to provide an ability to do so.

MAS team puts time and effort (free of charge, they are not paid for what they do) into things they want made first; you're free to make suggestions but you shouldn't expect your things to be on top of the pile for them to make, nor to expect it will be even implemented in foreseeable future.

You have some options:
* You suggest and wait, optionally providing support with adding detail to suggestion, explaining the better way to do it, etc. This will take a real while before the idea gets picked up and implemented.
* You work on it yourself and contribute to base mod, like 100+ contributors to this mod did. So far mainly topics and other stuff like be right backs and notifications, but no one stops you from creating something major. This will take some time to get rolled out because team cares about quality and reviews and tests stuff you add.
* You work on it yourself and make it a submod people will use on their own. You are on your own to maintain it, to support it and ensure people don't get their MAS experience spoiled by it. This will take as much time as you need (with amount of that time going all the way down to zero) because there's no one to tell you if it's a bad submod or not, you test it yourself, you review your code yourself, and it's up to you when it's ready or not.

Pick any of these three if you want new content added.

**Reply by ashyashton** [reply]
So once again: If you can code, you're in business. If not, you're screwed. But everyone is meant to act like that's a fair solution that doesn't screw over a ton of players.

**Comment by multimokia** [comment]
I see your points here. I'll keep this in mind to avoid suggesting submods personally unless Im actually aware of a submod and know it's one I personally trust to use (i.e. it is stable and designed with quality in mind)

That said I should note at least for peace of mind, while we of course don't officially support tech support for submods that may modify some internal MAS functions/value (which may cause data integrity issues), we are open to help making them and avoid data integrity issues, so in the event that we suggest a submod could be made, please understand we don't mean to defer, just that we likely don't have plans to add it officially, and if there is a desire for it from the community, that is _an_ option. To address your points on coding:

> Not everyone is capable of coding, but that doesn't invalidate the fact that they should feel represented, so the rebuttal of "make it yourself" doesn't really work either.

This is a fair point, and I reiterate we're open to questions and such regarding the API if anyone who's looking to start or get into submodding may have questions. Additionally there are some talented submodding groups out there who've made some excellent tools and guides to help get started over @Friends-Of-Monika thanks to their CI tools, guides, and the like. Likewise their communities are very much open to helping new programmers out too, so long as everyone's willing to learn. I get this may not be what you're looking for, but ultimately it's the best we can do given the nature of the situation, naturally there's a barrier to entry and of course, not everyone can clear it.

That said, we'll do what we can do avoid mentioning the creation of submods that are beyond the average user's (a person who's new to programming and/or python) capabilities. If it's simple enough, or it's documeted through the wiki, then I think it's fair game to suggest, but outside of that, I get where you come from.

**Comment by ashyashton** [comment]
See now that I fully agree with. I think it's totally fair to suggest if there's a submod you stand by, and would be willing to help troubleshoot on. It's just frustrating to see so many people say "well I know this is a complex new system in a coding environment you may not know, if you even know coding which most people don't, but do it yourself! if you truly cared, you would!" So thank you for taking what I've said under consideration. I really appreciate that.

Additionally, this is actually the first I've heard of Friends of Monika! I'll go check them out for sure. Until now, though, it was very much just "eh give it a shot and we'll be here if you have questions" without anyone being pointed in a clear direction for starting.
